### ISSUE TITLE
provide splash:on_response method

### ISSUE THREAD

**Issue by kmike** [scrapinghub/splash#230]
It should allow to register a callback called after each response is received, and allow to inspect the response, maybe change it and filter out based on its contents.

**Comment by kmike** [comment]
See also: https://github.com/scrapinghub/splash/issues/188, https://github.com/scrapinghub/splash/issues/231

**Comment by kmike** [comment]
https://github.com/scrapinghub/splash/pull/303 added `splash:on_response()` to Splash, but it is very limited: it is not possible to read response body, change the response or filter it out. It'd be also nice to be able to call async functions like `splash:wait()` from splash:on_response callbacks.

**Comment by chekunkov** [comment]
> but it is very limited: it is not possible to read response body, change the response or filter it out.

does it mean that it's impossible to remove some headers from response before rendering it? we need some way to remove `Content-Security-Policy`, `X-Content-Security-Policy` and `X-WebKit-CSP` headers from response...

**Comment by kmike** [comment]
Yes, it is not possible to remove headers. All necessary interfaces are private and not exposed by PyQT; I think that to make it work we may have to write custom C++ interfacing code.
